lib/s3middleware/tagging.ts

```diff
@@ -106,6 +117,54 @@ function _validateTags(tags: Array<{ Key: string[], Value: string[] }>) {
     return tagsResult;
 }
 
+/** validateTags - Validate tags, throwing an error if tags are invalid
+* @param tags - tags parsed from xml to be validated
+* @param tags[].Key - Name of the tag
+* @param tags[].Value - Value of the tag
+* @return tagsResult - return object tags on success
+* { key: value }; error on failure
+*/
+export function validateTags(tags: Array<BucketTag>) {
+    if (tags.length === 0) {
+        return {};
+    }
+    // Maximum number of tags per resource: 50
+    if (tags.length > 50) {
+        return errorBadRequestLimit50();
+    }
+
+    const tagsResult = {};
+    for (const tag of tags) {
+        if (!_validator.validateTagObjectStructure(tag)) {
```

Comment:
This will return false if key is falsey, negating the if below I believe. Unless there is some other condition I am missing.

Reply:
Right, that's an artifact from the original function where we were using an array.